services/api/lib/bulk.js

```diff
@@ -152,21 +152,44 @@ exports.register = function(server, options, done) {
   // 0th index and the value on that object keyed with 'id'
   var reachRegex = /^\$(\d+)\.(.*)$/;
 
+  var pipelineKeys = [
+    'id',
+    'projectId',
+    'pageId'
+  ];
+
+  function getIdKey(data) {
+    for (var i = 0; i < pipelineKeys.length; i++) {
+      if (data[pipelineKeys[i]]) {
+        return pipelineKeys[i];
+      }
+    }
+  }
+
+  function isReachString(value) {
+    return reachRegex.test(value);
+  }
+
+  function replaceValue(data, key, actionIndex, txResults) {
```

Comment:
it's not returned from the function... is it global?? (i assume not since it is passed into the func)